runtime/src/snapshot_utils.rs

```diff
@@ -1769,6 +1784,28 @@ fn rebuild_bank_from_snapshots(
     Ok(bank)
 }
 
+/// Verify that the snapshot's slot deltas are not corrupt/invalid
+fn verify_slot_deltas(
+    slot_deltas: &[BankSlotDelta],
+    bank_slot: Slot,
+) -> std::result::Result<(), VerifySlotDeltasError> {
+    for (slot, is_root, _) in slot_deltas {
+        // all entries should be roots
+        if !is_root {
```

Comment:
`ledger-tool create-snapshot` can legitimately create a opt-conf-ed slot's snapshot (i.e. with not rooted status delta) like for cluster restart, maybe?

even for that case, status deltas for such a snapshot are ensured to be `is_root == true` here?

could you confirm this?

Reply:
> Therefore my conclusion is that when ledger-tool creates snapshots, the its snapshot status cache will only contain status for roots. Does that sound right?

now that I believe this is true. thanks for extra trouble for explanation.

for future reference,  `bank_to_full_snapshot_archive` calls `Bank::squash()`, before calling `.root_slot_deltas()`. So, all parent and itself should have been marked as rooted at the time.